test/sagemaker_tests/pytorch/training/integration/sagemaker/test_neuron.py

```diff
@@ -17,23 +17,90 @@
 import sagemaker
 from sagemaker import utils
 from sagemaker.pytorch import PyTorch
-from ...integration import (neuron_allreduce_path, DEFAULT_TIMEOUT)
+from ...integration import (neuron_allreduce_path, neuron_mlp_path, DEFAULT_TIMEOUT)
 from ...integration.sagemaker.timeout import timeout
-from .... import invoke_pytorch_helper_function
+from retrying import retry
 
-RESOURCE_PATH = os.path.join(os.path.dirname(__file__), '..', '..', 'resources')
+def retry_if_value_error(exception):
+    """Return True if we should retry (in this case when it's an ValueError), False otherwise"""
+    return isinstance(exception, ValueError)
 
+@retry(
+    stop_max_attempt_number=360, wait_fixed=10000, retry_on_exception=retry_if_value_error,
+)
+def invoke_neuron_helper_function(ecr_image, sagemaker_regions, helper_function, helper_function_args):
+    """
+    Used to invoke SM job defined in the helper functions in respective test file. The ECR image and the sagemaker
+    session are passed explicitly depending on the AWS region.
+    This function will rerun for all SM regions after a defined wait time if capacity issues are seen.
+
+    :param ecr_image: ECR image in us-west-2 region
+    :param sagemaker_regions: List of SageMaker regions
+    :param helper_function: Function to invoke
+    :param helper_function_args: Helper function args
+
+    :return: None
+    """
+    from ..... import get_ecr_image_region, get_sagemaker_session, get_ecr_image
+
+    ecr_image_region = get_ecr_image_region(ecr_image)
+    for region in sagemaker_regions:
+        sagemaker_session = get_sagemaker_session(region)
+        # Reupload the image to test region if needed
+        tested_ecr_image = get_ecr_image(ecr_image, region) if region != ecr_image_region else ecr_image
+        try:
+            helper_function(tested_ecr_image, sagemaker_session, **helper_function_args)
+            return
+        except sagemaker.exceptions.UnexpectedStatusException as e:
+            if "CapacityError" in str(e):
+                raise ValueError("CapacityError: Retry.")
+            else:
+                raise e
+
+@pytest.mark.processor("neuron")
+@pytest.mark.model("unknown_model")
+@pytest.mark.neuron_test
+def test_neuron_allreduce_distributed(framework_version, ecr_image, sagemaker_regions, neuron_efa_instance_type):
+    function_args = {
+            'framework_version': framework_version,
+            'instance_type': neuron_efa_instance_type,
+            'instance_count': 2,
+        }
+    invoke_neuron_helper_function(ecr_image, sagemaker_regions, _test_neuron_allreduce_distributed, function_args)
```

Comment:
Same, as said above, we do not need the helper function, add the contents of `_test_neuron_allreduce_distributed` function here directly